DEVELOPMENT.md

```diff
@@ -1,3 +1,13 @@
+## Development Guidelines
+
+### Best Practices
+
+#### Working with Streams and Iterators
+
+**FluentIterable** from Google Guava is the established reference interface for operating on iterator and stream-like data structures in this codebase. It provides a fluent API for transforming, filtering, and manipulating collections in a functional programming style.
+
+While we also considered the Java Streams API we discourage its use because the `Spliterator` concept together with `tryAdvance` can sneakly lead to OOMs due to materializing full Iterators
```

Comment:
```suggestion
While we also considered the Java Streams API, we discourage its use because the `Spliterator` concept together with `tryAdvance` can sneakily lead to OOMs due to materializing full Iterators.
```